packages-internal/test-utils/src/describeConformance.tsx

```diff
@@ -264,7 +288,39 @@ function forEachSlot(
   });
 }
 
-function testSlotsProp(element: React.ReactElement<any>, getOptions: () => ConformanceOptions) {
+function testSlotsProp(
+  element: React.ReactElement<{
+    className: string;
+    classes: Record<string, string>;
+    slots: {
+      [x: string]:
+        | SlotTestingOptions['testWithComponent']
+        | keyof React.JSX.IntrinsicElements
+        | React.ForwardRefExoticComponent<
+            {
+              children: React.ReactNode;
+            } & React.RefAttributes<HTMLDivElement>
+          >;
+    };
+    components: {
```

Comment:
We should make `components` and `componentsProps` optional (`?`), as they're deprecated APIs and their usage is optional (with the `testLegacyComponentsProp` flag).

Is there a way for us to do that and silence the warnings that might appear inside the `testLegacyComponentsProp` tests?